pyomo/contrib/piecewise/tests/test_nested_inner_repn_gdp.py

```diff
@@ -0,0 +1,199 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2022
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import pyomo.common.unittest as unittest
+from pyomo.contrib.piecewise.tests import models
+import pyomo.contrib.piecewise.tests.common_tests as ct
+from pyomo.core.base import TransformationFactory
+from pyomo.environ import SolverFactory, Var, Constraint
+from pyomo.gdp import Disjunction, Disjunct
+from pyomo.core.expr.compare import assertExpressionsEqual
+
+
+# Test the nested inner repn gdp model using the common_tests code
+class TestTransformPiecewiseModelToNestedInnerRepnGDP(unittest.TestCase):
+    # Check one disjunct for proper contents. Disjunct structure should be
+    # identical to the version for the inner representation gdp
```

Comment:
It wouldn't really be appropriate for common_tests, since that file is also used for other transformations that are not inner representation-related. We could do this by adding moving the methods into a superclass between unittest.TestCase and the transformation test class, but I feel like that's unnecessarily complex to only house two duplicated methods. Well, we could start moving the test method boilerplate into the superclass too, but then it becomes inconsistent with the other transformations that have all their test methods duplicated.

Reply:
Maybe lets not convolute the class structure any more--I agree that that's not worth it, but could we move the methods themselves into functions in common_tests (or make a short `common_inner_repn_tests` script if you prefer), and just make this method call that function? I do this in GDP a lot: my convention is to name the function `check_whatever`, have its first argument be the test class (`self`) and then in the `test_whatever` method, I just call `check_whatever`. Sorry to be picky, but maintaining tests in code that could potentially change a lot is *really* hard, and I think having it be very clear to future us what we expect to be common will be nice. :)

Reply:
That works. I've moved the methods to a common_inner_repn_tests file.